src/legacy/index.ts

```diff
@@ -314,32 +312,42 @@ async function buildSubtree(
   const depInstanceIds = depGraph.getNodeDepsNodeIds(nodeId);
   if (!depInstanceIds || depInstanceIds.length === 0) {
     memoizationMap.set(nodeId, depTree);
-    return depTree;
+    return [depTree, false];
   }
 
   if (maybeDeduplicationSet) {
     if (maybeDeduplicationSet.has(nodeId)) {
       if (depInstanceIds.length > 0) {
         addLabel(depTree, 'pruned', 'true');
       }
-      return depTree;
+      return [depTree, false];
     }
     maybeDeduplicationSet.add(nodeId);
   }
 
+  let cyclePruned = false;
   for (const depInstId of depInstanceIds) {
+    if (ancestors.includes(depInstId)) {
+      cyclePruned = true;
+      addLabel(depTree, CYCLE_PRUNED_NODE_ID, 'true');
```

Comment:
Although weird, we already have a label for this: https://github.com/snyk/dep-graph/blob/master/src/core/types.ts#L47
```suggestion
      addLabel(depTree, 'pruned', 'cyclic');
```